apps/settings/js/panels/call/panel.js

```diff
@@ -0,0 +1,460 @@
+        }
+
+        this._bindPanelReadyEvent();
+        this._bindVisibilityChangeEvent();
+        this._bindCallForwardingMenuItemClickEvent();
+
+        this._addVoiceTypeChangeListeners();
+
+        this._initCallerId();
+        this._initCallWaiting();
+        this._initVoiceMailSettings();
+        this._initVoicePrivacyModeChangeEvent();
+        this._initVoicemailClickEvent();
+        this._initAlertLabelClickEvent();
+        this._initAlertCancelButtonClickEvent();
+
+        CallSettings.observe('isLocked', (isLocked) => {
+          this._enableTapOn('callerIdItem', !isLocked);
+          this._enableTapOn('callWaitingItem', !isLocked);
+        });
+
+        this._refreshCallSettingItems();
+      },
+
+      onBeforeShow: function(panel, options) {
+        if (options.cardIndex) {
+          this._cardIndex = options.cardIndex;
+          this._conn = this._conns[options.cardIndex];
+          CallSettings.cardIndex = this._cardIndex;
+        }
+
+        this._initFDNItem();
+        this._updateNetworkTypeLimitedItemsVisibility(
+          this._conn.voice && this._conn.voice.type);
+      },
+
+      _bindPanelReadyEvent: function() {
+        window.addEventListener('panelready', (evt) => {
+          switch (evt.detail.current) {
+            case '#call':
+              // No need to refresh the call settings items if navigated from
+              // panels not manipulating call settings.
+              if (evt.detail.previous === '#call-cfSettings' ||
+                evt.detail.previous === '#call-cbSettings') {
+                  return;
+              }
+              this._updateNetworkTypeLimitedItemsVisibility(
+                this._conn.voice && this._conn.voice.type);
+              this._refreshCallSettingItems();
+              break;
+          }
+        });
+      },
+
+      _bindVisibilityChangeEvent: function() {
+        document.addEventListener('visibilitychange', () => {
+          if (document.hidden) {
+            return;
+          }
+
+          switch (Settings.currentPanel) {
+            case '#call':
+              this._updateNetworkTypeLimitedItemsVisibility(
+                this._conn.voice && this._conn.voice.type
+              );
+              this._refreshCallSettingItems();
+              break;
+          }
+        });
+      },
+
+      _bindCallForwardingMenuItemClickEvent: function() {
+        this._elements.callForwardingAnchor.onclick = () => {
+          SettingsService.navigate('call-cfSettings', {
+            cardIndex: this._cardIndex
+          });
+        };
+      },
+
+      _initCallerId: function() {
+        // We listen for blur events so that way we set the CLIR mode once the
+        // user clicks on the OK button.
+        this._elements.rilCallerId.onblur = () => {
+          var clirMode = CallConstant.CLIR_MAPPING[
+            this._elements.rilCallerId.value];
+          this._conn.setCallingLineIdRestriction(clirMode).then(() => {
```

Comment:
I think these functions should also be included in `CallSettings` because we can only do them one at a time.